src/main.cpp

```diff
@@ -67,6 +71,13 @@ int main(int argc, char* argv[])
 
   const auto finish = std::chrono::high_resolution_clock::now();
   const double elapsedTime =
-      std::chrono::duration_cast<std::chrono::seconds>(finish - start).count();
+      std::chrono::duration_cast<std::chrono::duration<double>>(finish - start).count();
   std::cout << "Elapsed time is: " + std::to_string(elapsedTime) + "s." << std::endl;
+
+  // Test for correct output if the result_description is given in the input file
+  if (ExpectedPressure >= 0)
+  {
+    TEUCHOS_TEST_FOR_EXCEPTION(std::abs(pressure - ExpectedPressure) > ExpectedPressureTolerance,

```

Comment:
This is the wrong test macro. You are testing if the code raises a `std::runtime_error`. You would need something like [`TEST_FLOATING_EQUALITY`](https://docs.trilinos.org/dev/packages/teuchos/doc/html/group__Teuchos__UnitTestAssertMacros__grp.html#gaa60899c51ae9ffa9e76ca5cb41d7a0eb).

Side note: When implementing tests, always try to intentionally break the code and see if the test catches it. I assume in this case if you put the expected pressure to a wrong value the test will still pass.

Reply:
Okay. I will change the test macro. However, the current test macro worked without any issues. I did intentionally break the code with an incorrect expected value and it does catch it.